src/nyx/arrow_output_stream.cpp

```diff
@@ -0,0 +1,49 @@
+#ifdef USE_ARROW
+
+#include "arrow_output_stream.h"
+
+std::shared_ptr<ApacheArrowWriter> ArrowOutputStream::create_arrow_file(const std::string& arrow_file_type,
+                                                         const std::string& arrow_file_path,
+                                                         const std::vector<std::string>& header) {
+    
+    std::string arrow_file_type_upper = Nyxus::toupper(arrow_file_type);
+
+    if(arrow_file_path != "" && !fs::is_directory(arrow_file_path) && !(Nyxus::ends_with_substr(arrow_file_path, ".arrow") || Nyxus::ends_with_substr(arrow_file_path, ".feather") || Nyxus::ends_with_substr(arrow_file_path, ".parquet"))) {
+        throw std::invalid_argument("The arrow file path must end in \".arrow\"");
+    }
+
+    if (!(arrow_file_type_upper == "ARROW" || arrow_file_type_upper == "ARROWIPC" || arrow_file_type_upper == "PARQUET")) {
```

Comment:
How a user can specify via arrow_file_type_upper that the desired format is ".feather" ?